torch/_higher_order_ops/invoke_subgraph.py

```diff
@@ -94,13 +94,13 @@ def gen_schema(self, subgraph, identifier, *operands):
         schema_gen = HopSchemaGenerator(self)
         schema_gen.add_arg("subgraph", gm)
         schema_gen.add_arg("identifier", identifier)
-        example_inputs = [
-            n.meta["val"] if "val" in n.meta else n.meta["example_value"]
-            for n in gm.graph.find_nodes(op="placeholder")
-        ]
-        _, _, _, mutated_inputs, outputs = check_input_alias_and_mutation_return_ouputs(
-            gm, example_inputs
-        )
+        (
+            _,
+            _,
+            _,
+            mutated_inputs,
+            outputs,
+        ) = check_input_alias_and_mutation_return_outputs(gm, operands)
```

Comment:
Do you have some explanation about why this is better? I don't have a good sense of what we're doing here.